modules/clients/src/test/java/org/apache/ignite/qa/query/JdbcThinTimezoneTest.java

```diff
@@ -0,0 +1,369 @@
+            "SELECT tz, label, " +
+                "CAST(dateVal AS VARCHAR), " +
+                "CAST(timeVal AS VARCHAR), " +
+                "CAST(tsVal AS VARCHAR) " +
+                "FROM TZ_TEST", false));
+    }
+
+    /** */
+    @Test
+    public void testSelectWithConditions() throws Exception {
+        for (String tz : TIME_ZONES) {
+            insertObjectByLegacyApi(URL, TimeZone.getTimeZone(tz));
+            insertObjectByModernApi(URL, TimeZone.getTimeZone(tz));
+            insertLiteral(URL, TimeZone.getTimeZone(tz));
+        }
+
+        checkResults(selectResultsForAllTimeZones(
+            "SELECT tz, label, dateVal, timeVal, tsVal FROM TZ_TEST " +
+                "WHERE " +
+                "dateVal = CAST('2019-09-09' AS DATE) " +
+                "AND timeVal = CAST('09:09:09' AS TIME) " +
+                "AND tsVal = CAST('2019-09-09 09:09:09.909' AS TIMESTAMP)", true));
+
+        Map<String, List<String>> resMap = new HashMap<>();
+
+        for (String tz : TIME_ZONES) {
+            List<String> res = selectWithCondition(TimeZone.getTimeZone(tz));
+
+            res.sort(String::compareTo);
+
+            resMap.put(tz, res);
+        }
+
+        checkResults(resMap);
+    }
+
+    /** */
+    @Test
+    public void testDisableTimezone() throws Exception {
+        for (String tz : TIME_ZONES) {
+            insertObjectByLegacyApi(URL_TZ_DISABLE, TimeZone.getTimeZone(tz));
+            insertObjectByModernApi(URL_TZ_DISABLE, TimeZone.getTimeZone(tz));
+            insertLiteral(URL_TZ_DISABLE, TimeZone.getTimeZone(tz));
+        }
+
+        Map<String, List<String>> resMap = new HashMap<>();
+
+        for (String tz : TIME_ZONES) {
+            List<String> res = selectAndPrintMilliseconds("SELECT tz, label, dateVal, timeVal, tsVal FROM TZ_TEST",
+                TimeZone.getTimeZone(tz));
+
+            res.sort(String::compareTo);
+
+            resMap.put(tz, res);
+        }
+
+        checkResults(resMap);
+    }
+
+    /** */
+    @NotNull private Map<String, List<String>> selectResultsForAllTimeZones(String sql, boolean fetchDateObjs) throws Exception {
+        Map<String, List<String>> resMap = new HashMap<>();
+
+        for (String tz : TIME_ZONES) {
+            List<String> res = select(sql, TimeZone.getTimeZone(tz), fetchDateObjs);
+
+            res.sort(String::compareTo);
+
+            resMap.put(tz, res);
+        }
+
+        return resMap;
+    }
+
+    /** */
+    private void checkResults(Map<String, List<String>> resMap) {
+        for (String tz : TIME_ZONES) {
+            final List<String> res = resMap.get(tz);
+
+            assertFalse(res.isEmpty());
+
+            resMap.forEach((key, value) -> {
+                if (!value.equals(res)) {
+                    for (String s : res)
+                        System.err.println(tz + ", " + s);
+
+                    for (String s : value)
+                        System.err.println(key + ", " + s);
+
```

Comment:
should we remove the debug code?

Reply:
I guess this is not debug print. This is the extended assertion info. I think it doesn't make sense. 
We print results that doesn't pass the check.
The test doesn't print any results when all passes.